### ISSUE TITLE
AttributeError: 'BooleanFalse' object has no attribute 'name'

### ISSUE THREAD

**Issue by frnidito** [sympy/sympy#19736]
It seems that since version 1.6 the `BooleanFalse` is missing the attribute `name`, the same code was working and passing all the unittests in our codebase until version 1.5.1 (the last we used).

Such breaking change does not seem to be mentioned in the release notes for 1.6: was the remove intentional?

**Comment by oscarbenjamin** [comment]
I don't think that BooleanFalse has ever had a name attribute at least not in 1.5.1, 1.4, ...

I expect the problem you are seeing is due to something else. Somehow you have ended up with `S.false` instead of a symbol I guess.

**Comment by frnidito** [comment]
I can consistently reproduce this without changing any code: with 1.5.1 the test passes and with 1.6 it does not with the exception in the title

**Comment by oscarbenjamin** [comment]
Yes, but the problem is something else and you haven't provided any information for anyone to help you identify it
